### PR TITLE
Python: SignedTransactionPayload changes

### LINKED ISSUE #1486: Python: SignedTransaction payload changes
https://github.com/iotaledger/iota-sdk/pull/1483

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Removes the transaction essence from python. Renames `TransactionPayload` and `Transaction`.

## Links to any relevant issues

Closes #1486

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Failing CI

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Failing CI

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

File: bindings/python/examples/how_tos/nft_collection/00_mint_issuer_nft.py [hunk 1]
```diff
@@ -28,21 +28,21 @@
...
+    tx.transaction_id)
 
 print(
     f'Block sent: {os.environ["EXPLORER_URL"]}/block/{block_id}')
 
-essence = transaction.payload["essence"]
+transaction = tx.payload.transaction
 
-for outputIndex, output in enumerate(essence["outputs"]):
+for outputIndex, output in enumerate(transaction["outputs"]):
```
Comment (thibault-martinez) [inline bindings/python/examples/how_tos/nft_collection/00_mint_issuer_nft.py]: Can we also remove `["outputs"]`?

Comment (thibault-martinez) [reply bindings/python/examples/how_tos/nft_collection/00_mint_issuer_nft.py]: I mean, use `.outputs` instead

Comment (DaughterOfMars) [reply bindings/python/examples/how_tos/nft_collection/00_mint_issuer_nft.py]: done

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Pull request merged**